cranelift-simplejit/src/backend.rs

```diff
@@ -531,6 +548,22 @@ fn lookup_with_dlsym(name: &str) -> *const u8 {
     }
 }
 
+impl SimpleJITMemoryHandle {
+    /// Free memory allocated for code and data segments of compiled functions.
+    ///
+    /// # Safety
+    ///
+    /// Because this function invalidates any pointers retrived from the
+    /// corresponding module, it should only be used when none of the functions
+    /// from that module are currently executing and none of the`fn` pointers
+    /// are called afterwards.
+    pub unsafe fn free_memory(&mut self) {
```

Comment:
Simplejit itself returns only raw pointer. Converting them to a fnptr means that you assert that the memory is not yet freed. This means that calling this function is safe, just using the raw pointers is unsafe.

Reply:
However keeping it unsafe could prevent suprises.

Reply:
I think keeping it unsafe helps signify that you should not use it unless you know you need to. Also, as I mentioned before, the cast to `fn` is almost required to use `SimpleJIT`, so it probably should be treated like returning `fn`.